arch/xtensa/src/esp32s3/esp32s3_serial.c

```diff
@@ -687,7 +675,7 @@ static bool esp32s3_txempty(struct uart_dev_s *dev)
 
 static void esp32s3_send(struct uart_dev_s *dev, int ch)
 {
-  esp32s3_lowputc_send_byte(dev->priv, ch);
+  esp32s3_lowputc_send_byte(dev->priv, (char)ch);
```

Comment:
I do not think that this is needed

Reply:
This is for silencing clang-tidy's *bugprone-narrowing-conversions* warning:
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-narrowing-conversions.html

Reply:
So you are compiling this code with C++ compiler?

Reply:
No, I am just using clang-tidy as a static code analyzer.
This warning is a minor one, I agree with you that is not actually needed and will probably have no impact on the generated code.

Reply:
yes. NP with it